p2panda-net-next/src/actors/streams/ephemeral.rs

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Ephemeral streams actor.
+//!
+//! This actor forms a coordination layer between the external API and the gossip sub-systems.
+//!
+//! No actors are spawned or respawned by the ephemeral streams actor, that role is carried out
+//! by the stream supervisor.
+use std::collections::HashMap;
+
+/// Ephemeral streams actor name.
+pub const EPHEMERAL_STREAMS: &str = "net.streams.ephemeral";
+
+use ractor::thread_local::{ThreadLocalActor, ThreadLocalActorSpawner};
+use ractor::{ActorProcessingErr, ActorRef, RpcReplyPort, call, cast};
+use tokio::sync::broadcast::Sender as BroadcastSender;
+use tokio::sync::mpsc::Sender;
+
+use crate::TopicId;
+use crate::actors::ActorNamespace;
+use crate::actors::gossip::ToGossip;
+use crate::actors::sync::SyncManager;
+use crate::network::{FromNetwork, ToNetwork};
+use crate::streams::ephemeral::{EphemeralStream, EphemeralSubscription};
+
+pub enum ToEphemeralStreams {
+    /// Create an ephemeral stream for the topic ID and return a publishing handle.
+    Create(TopicId, RpcReplyPort<EphemeralStream>),
+
+    /// Return an ephemeral subscription handle for the given topic ID.
+    Subscribe(TopicId, RpcReplyPort<Option<EphemeralSubscription>>),
+
+    /// Close all ephemeral streams for the given topic ID.
+    Close(TopicId),
+
+    /// Return `true` if there are any active ephemeral streams for the given topic ID.
+    IsActive(TopicId, RpcReplyPort<bool>),
```

Comment:
I realised that we only want to trigger a gossip unsubscribe from the eventually consistent streams actor if there isn't any active _ephemeral_ stream. This allows us to check for active ephemeral streams when an eventually consistent stream is closed and when a sync manager actor fails.